eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelayer.scala

```diff
@@ -158,14 +158,19 @@ class NodeRelayer(nodeParams: NodeParams, relayer: ActorRef, router: ActorRef, c
     val paymentCfg = SendPaymentConfig(paymentId, paymentId, None, paymentHash, payloadOut.amountToForward, payloadOut.outgoingNodeId, upstream, None, storeInDb = false, publishEvent = false, Nil)
     val routeParams = computeRouteParams(nodeParams, upstream.amountIn, upstream.expiryIn, payloadOut.amountToForward, payloadOut.outgoingCltv)
     payloadOut.invoiceFeatures match {
-      case Some(_) =>
-        log.debug("relaying trampoline payment to non-trampoline recipient")
+      case Some(features) =>
         val routingHints = payloadOut.invoiceRoutingInfo.map(_.map(_.toSeq).toSeq).getOrElse(Nil)
-        // TODO: @t-bast: MPP is disabled for trampoline to non-trampoline payments until we improve the splitting algorithm for nodes with a lot of channels.
-        val payFSM = spawnOutgoingPayFSM(paymentCfg, multiPart = false)
-        val finalPayload = Onion.createSinglePartPayload(payloadOut.amountToForward, payloadOut.outgoingCltv, payloadOut.paymentSecret)
-        val payment = SendPayment(payloadOut.outgoingNodeId, finalPayload, nodeParams.maxPaymentAttempts, routingHints, Some(routeParams))
-        payFSM ! payment
+        payloadOut.paymentSecret match {
```

Comment:
Why do we have this equivalence between "non-trampoline next node" and "existence of payment secret"?

Reply:
We don't :)
The code here starts by looking at whether we have `invoiceFeatures` available in the onion. This field will only be there when a phoenix node asks us to relay to a non-trampoline node.
Then we check the `paymentSecret` to see whether we can use MPP or not to reach that non-trampoline destination node.
Does that make more sense?

Reply:
Yes but definitely probably deserves a comment

Reply:
Done in https://github.com/ACINQ/eclair/pull/1439/commits/da4f8694edbfc83cfcb3e369079930aac028c748